src/app/file/webp_format.cpp

```diff
@@ -0,0 +1,390 @@
+#include "app/document.h"
+#include "app/file/file.h"
+#include "app/file/file_format.h"
+#include "app/file/format_options.h"
+#include "app/ini_file.h"
+#include "base/file_handle.h"
+#include "doc/doc.h"
+
+#include "generated_webp_options.h"
+
+#include <cstdio>
+#include <cstdlib>
+#include <algorithm>
+#include <map>
+
+
+//include webp librarys
+#include <webp/decode.h>
+#include <webp/encode.h>
+
+namespace app {
+
+using namespace base;
+
+class WebPFormat : public FileFormat {
+  // Data for WEBP files TODO make this better OOP like GIF_OPTIONS
+  class WebPOptions : public FormatOptions {
+  public:
+    WebPOptions(): lossless(1), quality(75), method(6), image_hint(WEBP_HINT_DEFAULT), image_preset(WEBP_PRESET_DEFAULT) {};
+    int lossless;           // Lossless encoding (0=lossy(default), 1=lossless).
+    float quality;          // between 0 (smallest file) and 100 (biggest)
+    int method;             // quality/speed trade-off (0=fast, 9=slower-better)
+    WebPImageHint image_hint;  // Hint for image type (lossless only for now).
+    WebPPreset image_preset;  // Image Preset for lossy webp.
+  };
+
+  const char* onGetName() const { return "webp"; }
+  const char* onGetExtensions() const { return "webp"; }
+  int onGetFlags() const {
+    return
+      FILE_SUPPORT_LOAD |
+      FILE_SUPPORT_SAVE |
+      FILE_SUPPORT_RGB |
+      FILE_SUPPORT_RGBA |
+      FILE_SUPPORT_SEQUENCES |
+      FILE_SUPPORT_GET_FORMAT_OPTIONS;
+  }
+
+  bool onLoad(FileOp* fop) override;
+#ifdef ENABLE_SAVE
+  bool onSave(FileOp* fop) override;
+#endif
+
+base::SharedPtr<FormatOptions> onGetFormatOptions(FileOp* fop) override;
+};
+
+FileFormat* CreateWebPFormat()
+{
+  return new WebPFormat;
+}
+
+const std::pair<VP8StatusCode, std::string> dec_error_map_data[] = {
+    std::make_pair(VP8_STATUS_OK, ""),
+    std::make_pair(VP8_STATUS_OUT_OF_MEMORY, "out of memory"),
+    std::make_pair(VP8_STATUS_INVALID_PARAM, "invalid parameters"),
+    std::make_pair(VP8_STATUS_BITSTREAM_ERROR, "bitstream error"),
+    std::make_pair(VP8_STATUS_UNSUPPORTED_FEATURE, "unsupported feature"),
+    std::make_pair(VP8_STATUS_SUSPENDED, "suspended"),
+    std::make_pair(VP8_STATUS_USER_ABORT, "user aborted"),
+    std::make_pair(VP8_STATUS_NOT_ENOUGH_DATA, "not enough data")
+};
+
+const std::map<VP8StatusCode, std::string> WebPDecodingErrorMap(dec_error_map_data, dec_error_map_data + sizeof dec_error_map_data / sizeof dec_error_map_data[0]);
+
+bool WebPFormat::onLoad(FileOp* fop)
+{
+  FileHandle handle(open_file_with_exception(fop->filename, "rb"));
+  FILE* fp = handle.get();
+
+  long len;
+  uint8_t* buf = NULL;
+
+  if (fseek(fp, 0, SEEK_END) != 0) {
+    fop_error(fop, "Error while getting WebP file size for %s\n", fop->filename.c_str());
+    return false;
+  }
+
+  len = ftell(fp);
+  rewind(fp);
+
+  if (len < 4) {
+    fop_error(fop, "%s is corrupt or not a WebP file\n", fop->filename.c_str());
+    return false;
+  }
+
+  buf = (uint8_t*) malloc(len);
```

Comment:
You can use [base_malloc()](https://github.com/aseprite/aseprite/blob/master/src/app/file/jpeg_format.cpp#L153) or a [std::vector](https://github.com/aseprite/aseprite/blob/master/src/app/file/gif_format.cpp#L1108) directly (the last one is preferred, you don't need to worry about base_free()).